pint/quantity.py

```diff
@@ -1302,62 +1315,56 @@ def __bool__(self):
 
-    # NumPy Support
-    __radian = 'radian'
-    __same_units = 'equal greater greater_equal less less_equal not_equal arctan2'.split()
-    #: Dictionary mapping ufunc/attributes names to the units that they
-    #: require (conversion will be tried).
-    __require_units = {'cumprod': '',
-                       'arccos': '', 'arcsin': '', 'arctan': '',
-                       'arccosh': '', 'arcsinh': '', 'arctanh': '',
-                       'exp': '', 'expm1': '', 'exp2': '',
-                       'log': '', 'log10': '', 'log1p': '', 'log2': '',
-                       'sin': __radian, 'cos': __radian, 'tan': __radian,
-                       'sinh': __radian, 'cosh': __radian, 'tanh': __radian,
-                       'radians': 'degree', 'degrees': __radian,
-                       'deg2rad': 'degree', 'rad2deg': __radian,
-                       'logaddexp': '', 'logaddexp2': ''}
-
-    #: Dictionary mapping ufunc/attributes names to the units that they
-    #: will set on output.
-    __set_units = {'cos': '', 'sin': '', 'tan': '',
-                   'cosh': '', 'sinh': '', 'tanh': '',
-                   'log': '', 'exp': '',
-                   'arccos': __radian, 'arcsin': __radian,
-                   'arctan': __radian, 'arctan2': __radian,
-                   'arccosh': __radian, 'arcsinh': __radian,
-                   'arctanh': __radian,
-                   'degrees': 'degree', 'radians': __radian,
-                   'expm1': '', 'cumprod': '',
-                   'rad2deg': 'degree', 'deg2rad': __radian}
-
-    #: List of ufunc/attributes names in which units are copied from the
-    #: original.
-    __copy_units = 'compress conj conjugate copy cumsum diagonal flatten ' \
-                   'max mean min ptp ravel repeat reshape round ' \
-                   'squeeze std sum swapaxes take trace transpose ' \
-                   'ceil floor hypot rint ' \
-                   'add subtract ' \
-                   'copysign nextafter trunc ' \
-                   'frexp ldexp modf modf__1 ' \
-                   'absolute negative remainder fmod mod'.split()
-
-    #: Dictionary mapping ufunc/attributes names to the units that they will
-    #: set on output. The value is interpreted as the power to which the unit
-    #: will be raised.
-    __prod_units = {'var': 2, 'prod': 'size', 'multiply': 'mul',
-                    'true_divide': 'div', 'divide': 'div', 'floor_divide': 'div',
-                    'remainder': 'div',
-                    'sqrt': .5, 'square': 2, 'reciprocal': -1}
-
-    __skip_other_args = 'ldexp multiply ' \
-                        'true_divide divide floor_divide fmod mod ' \
-                        'remainder'.split()
-
-    __handled = tuple(__same_units) + \
-                tuple(__require_units.keys()) + \
-                tuple(__prod_units.keys()) + \
-                tuple(__copy_units) + tuple(__skip_other_args)
+    # NumPy function/ufunc support
+    __array_priority__ = 17
```

Comment:
While [the NumPy docs](https://docs.scipy.org/doc/numpy/reference/arrays.classes.html#numpy.class.__array_priority__) imply that it is no longer used for ufuncs when `__array_ufunc__` is in use, Dask, Sparse, and CuPy still all seem to declare it, so I thought it best to maintain it here. 

17 is simply the value Pint had previously. This puts it above ndarray (10), Dask (11), Sparse (12), and masked arrays (15), which is all good. However, CuPy is unfortunately at 100, so this may need to be increased if Pint Quantities should be above CuPy arrays. Would you want to increment this value now, or leave it at the current value of 17 and discuss any possible changes in #924?

Reply:
Similar, to my previous comment. I think we should merge this ASAP to increase the base of tester. Regarding the number, I have two (not contradictory) options:
1. Kindly Suggest to CuPy to change the number
2. Get this value from an environmental variable (defaulting to 17). We should do this is in a way that is copyable by other packages (e.g. Naming the variable `PINT_NUMPY_ARRAY_PRIORITY` or similar)